pkg/querier/queryrange/roundtrip.go

```diff
@@ -1233,3 +1299,80 @@ func NewDetectedFieldsTripperware(
 		return NewDetectedFieldsHandler(limitedHandler, logHandler, limits)
 	}), nil
 }
+
+// NewVariantsTripperware creates a new frontend tripperware responsible for handling queries with multiple variants for a single
+// selector.
+func NewVariantsTripperware(
+	cfg Config,
+	engineOpts logql.EngineOpts,
+	log log.Logger,
+	limits Limits,
+	schema config.SchemaConfig,
+	merger base.Merger,
+	iqo util.IngesterQueryOptions,
+	_ cache.Cache,
+	_ base.CacheGenNumberLoader,
+	_ bool,
+	_ base.Extractor,
+	metrics *Metrics,
+	indexStatsTripperware base.Middleware,
+	metricsNamespace string,
+) (base.Middleware, error) {
+	return base.MiddlewareFunc(func(next base.Handler) base.Handler {
```

Comment:
As part of this PR, can we just return an error if the tripperware is visited?
This code is completely untested anyway

Reply:
sure, I'm fine with that :)